src/raster/array.jl

```diff
@@ -1,89 +1,133 @@
+import DiskArrays: eachchunk, GridChunks
+import DiskArrays
 const AllowedXY = Union{Integer,Colon,AbstractRange}
 const AllowedBand = Union{Integer,Colon,AbstractArray}
 
 
-# AbstractRasterBand indexing
+"""
+    struct RasterDataset{T} <: AbstractDiskArray{T,3}
+
+This data structure is returned by the `readraster` function and
+is a wrapper for a GDAL dataset. This wrapper is to signal the
+user that the dataset should be treated as a 3D AbstractArray
+where the first to dimensions correspond to latitude and longitude
+and the third dimension correpsonds to different raster bands.
+
+Constructing a RasterDataset will error if the raster bands do not
+have all the same size and a common element data type.
+"""
```

Comment:
Might be good to add something to the effect of: "As it is a wrapper around a GDAL Dataset, it supports the usual raster methods for a GDAL Dataset such as `getgeotransform`, `nraster`, `getband`, `getproj`, `width`, and `height`. As it is also a subtype of AbstractDiskArray{T,3}, it supports the following additional methods: `eachchunk`, etc. This satisfies the DiskArray interface, allowing us to be able to index into it like we would an array."

Reply:
Thanks, I added the text to the doc string